doc/iterations/2023.12/shapes/handles.adoc

```diff
@@ -0,0 +1,65 @@
+- handles serve as target point for the creation and reconnection of edges, but they are small and relatively hard to point to.
+
+The image below illustrates these issues:
+
+image::images/handles_initial_state.png[How handles are currently displayed on React Flow Diagram]
+
+1. The "black dots" visibles one the four sides of all nodes.
+2. Including nodes which can not actually be the source or target of edges.
+3. When two nodes (Wifi and DataSource2) point to the same side of another node (the processor), their extermity are "merged" into the same point and they are indistinguishable.
+
+== Key Result
+
+1. Unless the end-user is explicitly interacting with a node or edge, the handles used to create new edges or reconnect existing ones should not be visible.
+2. When a node has multiple edge extremities (including from the same "self loop" edge) pointing to one of its side, they are all clearly separated and spaced evenly across the node side.
+3. When creating a new edge or reconnecting an existing one, the user can point anywhere inside the bounding box of the target node.
+4. When creating a new edge or reconnecting an existing one, nodes which are incompatible/compatible/selected as the target are made clearly distinct visually.
+
+== Solution
+
+We will distinguish three different kinds of handles:
+
+1. *edge creation handles*: placed at the same location as the current ones (on the middle of each side of a node) but only used to initiate the creation of new edges.
+These will only be present when a node is selected *and* can be the source of an edge creation tool.
+2. *edge reconnection handles*: placed at the extremity of edges, these can be used to initiate the reconnection of an existing edge.
+3. *edge target handles*: these are invisible and cover the whole area of a node, but are required by React Flow to serve as the target zone for both edge creation and reconnection.
+
+Handles should only be visible when relevant:
+
+- *edge creation handles* are only visible on nodes which can have incoming or outgoing edges *and* are selected (just hovering on a node does _not_ make the handles visible);
+- *edge reconnection handles* are only visible when the corresponding edge is selected;
+- *edge target handles* are never visible, they exist only to determine the active zone for selecting the target of an edge creation or reconnect.
+
+If multiple edges point to the same side of a given node,  they should point to locations which are evenly spaced on the side, distinct from each other.
+This includes the case of "self loops" when both endpoints of an edge are the same face of the same node: they should be distinct.
+
+Style and feedback:
+
+* when a handle is visible (because the corresponding node or edge is selected), it is highlighted in a distinct color when hovered.
```

Comment:
The ConnectionHandles are visible when the associated edge is selected and the CreationHandles when a node with edge creation tool is selected.

Reply:
Reworded to make it clearer.